pkg/cloud/tags.go

```diff
@@ -16,30 +16,107 @@ limitations under the License.
 
 package cloud
 
+import (
+	infrav1 "github.com/aws/cluster-api-provider-cloudstack/api/v1beta1"
+	"strings"
+)
+
 type TagIface interface {
-	AddNetworkTags(string, map[string]string) error
-	GetNetworkTags(string) (map[string]string, error)
-	DeleteNetworkTags(string, map[string]string) error
+	AddClusterTag(resourceType ResourceType, resourceID string, csCluster *infrav1.CloudStackCluster, addCreatedByCAPCTag bool) error
```

Comment:
Gurd... Rebasing my branch on yours is going to stink. Sigh... oh well.

Reply:
Sorry.